extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergCatalogAdapter.java

```diff
@@ -188,10 +187,22 @@ private Table readTableInternal(
         // Load the partitioning schema
         final org.apache.iceberg.PartitionSpec partitionSpec = table.spec();
 
-        // Use the user-supplied defininition or the Iceberg schema for the table definition.
-        final TableDefinition tableDef = instructions.tableDefinition().isPresent()
-                ? instructions.tableDefinition().get()
-                : fromSchema(schema, partitionSpec, instructions);
+        // Get the user supplied table definition and column rename map (if any).
+        final TableDefinition userTableDef = instructions.tableDefinition().orElse(null);
+
+        // Get the table definition from the schema (potentially limited by the user supplied table definition and
+        // applying column renames).
+        final TableDefinition icebergTableDef =
+                fromSchema(schema, partitionSpec, userTableDef, instructions.columnRenameMap());
+
+        // If the user supplied a table definition, make sure it's fully compatible.
+        final TableDefinition tableDef;
+        if (userTableDef != null) {
+            tableDef = icebergTableDef.checkMutualCompatibility(userTableDef);
```

Comment:
If we do it this way (not mutual), we allow the user def to have *extra* columns (this is good).
```suggestion
            tableDef = icebergTableDef.checkCompatibility(userTableDef);
```